pkg/promutil/test/multiprojects_metric_scences.go

```diff
@@ -0,0 +1,538 @@
+package main
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/hanfei1991/microcosm/pkg/promutil"
+	"github.com/hanfei1991/microcosm/pkg/tenant"
+	"github.com/pingcap/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func main() {
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		log.L().Info("Start http listen on :8083")
+		http.ListenAndServe(":8083", nil)
+	}()
+
+	go func() {
+		defer wg.Done()
+		log.L().Info("Start scenarios simulator")
+		simulator(wg)
```

Comment:
If a `WaitGroup` is explicitly passed into functions, it must be done by pointer, such as discussion: https://youtrack.jetbrains.com/issue/GO-7472/WaitGroup-inspection-when-passing-wg-not-as-pointer